File: src/main/java/com/example/cqrs/command/dto/read/AccountDetailResponse.java
package com.example.cqrs.command.dto.read;

import com.example.cqrs.command.entity.AccountWrite;
import lombok.*;

/**
 * 계좌 상세 정보에 대한 Response DTO입니다.
 */
@AllArgsConstructor(access = AccessLevel.PRIVATE)
@NoArgsConstructor(access = AccessLevel.PRIVATE)
@Builder
@Getter
public class AccountDetailResponse {

    private String accountId;
    private double balance;

    // factory method
    public static AccountDetailResponse from(AccountWrite accountWrite) {
        return AccountDetailResponse.builder()
                .accountId(accountWrite.getAccountId())
                .balance(accountWrite.getBalance())
                .build();
    }

}

File: src/main/java/com/example/cqrs/command/dto/read/AccountDetailResponse.kt
package com.example.cqrs.command.dto.read

data class AccountDetailResponse(
    val accountId: String,
    val balance: Double
) {
    companion object {
        fun from(accountId: String, balance: Double): AccountDetailResponse {
            return AccountDetailResponse(accountId, balance)
        }
    }
}
